src/StructuredContentList.js

```diff
@@ -1,307 +1,336 @@
+      shouldLoadMoreBeVisible: true,
+      hasErrorGettingEntries: false,
+      filters: this.props.filters
+        ? this.props.filters.map(filter => {
+            filter.options = []; // ensure options
 
-    buildFilterQueryString() {
-        const activeFilters = this.getActiveFilters();
-        const appliedFilterQuery = this.getAppliedFiltersQuery();
-        let queryParams = [];
+            return filter;
+          })
+        : []
+    };
 
-        if (activeFilters) {
-            queryParams.push(`filters=${activeFilters}`);
-        }
-        if (appliedFilterQuery) {
-            queryParams.push(`$filter=${appliedFilterQuery}`);
-        }
+    this.onChange = this.onChange.bind(this);
+    this.getFilterOptions = this.getFilterOptions.bind(this);
+    this.getActiveFilters = this.getActiveFilters.bind(this);
+    this.getAppliedFiltersQuery = this.getAppliedFiltersQuery.bind(this);
 
-        return queryParams.join('&');
-    }
+    this.onLoadMore = this.onLoadMore.bind(this);
+  }
 
-    buildQueryString() {
-        const queryString = this.buildFilterQueryString();
-        const activePage = this.state.activePage;
+  buildFilterQueryString() {
+    const activeFilters = this.getActiveFilters();
+    const appliedFilterQuery = this.getAppliedFiltersQuery();
+    let queryParams = [];
 
-        return queryString
-            ? `?page=${activePage}&${queryString}`
-            : `?page=${activePage}`;
+    if (activeFilters) {
+      queryParams.push(`filters=${activeFilters}`);
     }
-
-    componentDidMount() {
-        this.getBlogEntries(() => {
-            this.setState({
-                isInitialized: true
-            });
-        });
+    if (appliedFilterQuery) {
+      queryParams.push(`$filter=${appliedFilterQuery}`);
     }
 
-    getActiveFilters() {
-        // filters are the distinct fields being returned, if there are some in the app, these should always be added to the query string
-        return this.state.filters.map(filter => filter.field);
-    }
+    return queryParams.join("&");
+  }
+
+  buildQueryString() {
+    const queryString = this.buildFilterQueryString();
+    const activePage = this.state.activePage;
+
+    return queryString
+      ? `?page=${activePage}&${queryString}`
+      : `?page=${activePage}`;
+  }
+
+  componentDidMount() {
+    this.getBlogEntries(() => {
+      this.setState({
+        isInitialized: true
+      });
+    });
+  }
 
-    getAppliedFilterList() {
-        const appliedFilters = { ...this.state.activeFilters };
-        return Object.keys(appliedFilters).map(appliedFilterKey => ({
-            key: appliedFilterKey,
-            value: appliedFilters[appliedFilterKey]
-        }));
+  getActiveFilters() {
+    // filters are the distinct fields being returned, if there are some in the app, these should always be added to the query string
+    return this.state.filters.map(filter => filter.field);
+  }
+
+  getTopOneElement(blogEntries) {
```

Comment:
think this can just be

```js
getTopOneElement(blogEntries) {
    return blogEntries.find(entry => entry.IsFeatured);
}
```

the service should return blog posts in the proper order. Checkout more about first [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)

Reply:
function name may be better suited as `getFeaturedBlogEntry`

Reply:
this should also allow you to get rid of lodash

Reply:
function name is changed